p2p/shred.md

```diff
@@ -0,0 +1,313 @@
+# Shred packets
+
+The Solana Turbine protocol propagates confirmed data (blocks) within peer-to-peer networks in the form of shred packets.
+
+Data shreds are created by splitting up block data.
+Code shreds are created by constructing erasure codes over the vector of data shreds.
+
+All shreds are signed by the block producer.
+
+## Revisions
+
+Multiple revisions of the shred structures exist.
+The shred revision is not explicitly encoded.
+
+### v1: Genesis revision
+
+Solana mainnet-beta genesis launched with support for code and data shreds with the legacy authentication mechanism.
+
+### v2: Explicit data sizes
+
+Breaking change to the data shred header adding a new size field at offset `0x56`.
+
+The shred paylod shifts from offset `0x56` to `0x58`.
+
+### v3: Merkle authentication mechanism
+
+Introduction of two additional shred variants with the Merkle authentication scheme.
+
+## Protocol
+
+### Transport
+
+Shreds are transmitted via unidirectional streams.
```

Comment:
Shred transmission is definitely unidirectional for initial turbine broadcast, but repairs involve a node explicitly asking to be sent specific shreds.

I'm wondering if we should drop the Transport section and save that for a separate spec on turbine? Ie this spec could cover the shreds (layout, data v coding, merkle, etc), and then turbine doc could reference this spec saying turbine acts on shreds?

If you agree with this sentiment, I would say to leave in the note in `Packet Size` section as that provides the justification for why that constant is that value.